crates/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -798,18 +816,41 @@ impl<'interner> Monomorphizer<'interner> {
     /// of field elements to/from JSON. The type metadata attached in this method
     /// is the serialized `AbiType` for the argument passed to the function.
     /// The caller that is running a Noir program should then deserialize the `AbiType`,
-    /// and accurately decode the list of field elements passed to the foreign call.  
-    fn append_abi_arg(&self, hir_argument: &HirExpression, arguments: &mut Vec<ast::Expression>) {
+    /// and accurately decode the list of field elements passed to the foreign call.
+    fn append_abi_arg(
+        &mut self,
+        hir_argument: &HirExpression,
+        arguments: &mut Vec<ast::Expression>,
+    ) {
         match hir_argument {
             HirExpression::Ident(ident) => {
                 let typ = self.interner.id_type(ident.id);
-                let typ = typ.follow_bindings();
-
-                let abi_type = typ.as_abi_type();
-                let abi_as_string =
-                    serde_json::to_string(&abi_type).expect("ICE: expected Abi type to serialize");
-
-                arguments.push(ast::Expression::Literal(ast::Literal::Str(abi_as_string)));
+                let typ: Type = typ.follow_bindings();
+                match &typ {
+                    // A format string has many different possible types that need to be handled.
+                    // Loop over each element in the format string to fetch each type's relevant metadata
+                    Type::FmtString(_, element_types) => {
+                        for typ in element_types {
+                            let abi_type = typ.as_abi_type();
+                            let abi_as_string = serde_json::to_string(&abi_type)
+                                .expect("ICE: expected Abi type to serialize");
+
+                            arguments
+                                .push(ast::Expression::Literal(ast::Literal::Str(abi_as_string)));
+                        }
+                        // The caller needs information as to whether it is handling a format string or a single type
+                        arguments.push(ast::Expression::Literal(ast::Literal::Bool(true)));
```

Comment:
I think we shouldn't allow fmt strings at all in abi args

Reply:
This isn't really a format string in the ABI, but rather a distinction to the caller of the println foreign call on how to deal with the list of field element inputs they have been provided. I foresee this changing from a bool to an encoding on some type of enumeration once we add format parameters (do not format at all, default format, pretty-print format, hexadecimal format, etc.). Right now it is essentially an extra field as part of the encoding that `nargo` uses on how to handle the inputs it has been provided (https://github.com/noir-lang/noir/blob/a97f251cc8aeb68f9c62bc90b406c66cb30ac90e/crates/nargo/src/ops/foreign_calls.rs#L70).